### PR TITLE
Fix Incorrectly Typed Goals

### PR DESCRIPTION
**Pull request by tahmid-23**
I wasn't sure about whether to rename SKELETON_MELEE to ABSTRACT_SKELETON_MELEE since it's API but at the same time this is not what the test would produce

### CODE REVIEW

File: patches/server/0293-Don-t-allow-digging-into-unloaded-chunks.patch [hunk 1]
```diff
@@ -63,20 +63,16 @@ index 315dad4789f5f2582ee9b4fc176affd1f57537ef..482eebf61944add9f1ad6abea6124a9f
...
              case ABORT_DESTROY_BLOCK:
              case STOP_DESTROY_BLOCK:
 -                this.player.gameMode.handleBlockBreakAction(blockposition, packetplayinblockdig_enumplayerdigtype, packet.getDirection(), this.player.level.getMaxBuildHeight());
 +                // Paper start - Don't allow digging in unloaded chunks
-+                // Copied from ServerPlayerGameMode#handleBlockBreakAction
-+                double d0 = this.player.getX() - ((double) blockposition.getX() + 0.5D);
-+                double d1 = this.player.getY() - ((double) blockposition.getY() + 0.5D) + 1.5D;
-+                double d2 = this.player.getZ() - ((double) blockposition.getZ() + 0.5D);
-+                double blockDistanceSquared = d0 * d0 + d1 * d1 + d2 * d2;
++                double blockDistanceSquared = this.player.distanceToSqr(blockposition.getX() + 0.5D, blockposition.getY() + 0.5D + 1.5D, blockposition.getZ() + 0.5D); // Copied from ServerPlayerGameMode#handleBlockBreakAction
```
Comment (Proximyst) [inline patches/server/0293-Don-t-allow-digging-into-unloaded-chunks.patch]: why

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: patches/server/0293-Don-t-allow-digging-into-unloaded-chunks.patch (hunk 1 above)
Comment (tahmid-23) [reply patches/server/0293-Don-t-allow-digging-into-unloaded-chunks.patch]: IntelliJ oops

File: patches/server/0431-Implement-Mob-Goal-API.patch [hunk 2]
```diff
@@ -50,102 +50,14 @@ index 0000000000000000000000000000000000000000..dfc026d183adab1dde5942f36e7a281b
...
-+import org.bukkit.entity.WaterMob;
-+import org.bukkit.entity.Witch;
-+import org.bukkit.entity.Wither;
-+import org.bukkit.entity.WitherSkeleton;
-+import org.bukkit.entity.Wolf;
-+import org.bukkit.entity.Zoglin;
-+import org.bukkit.entity.Zombie;
-+import org.bukkit.entity.ZombieHorse;
-+import org.bukkit.entity.ZombieVillager;
++import org.bukkit.entity.*;
```
Comment (Proximyst) [inline patches/server/0431-Implement-Mob-Goal-API.patch]: no

**Review by Proximyst — CHANGES_REQUESTED** [review]
What does this even fix?

File: patches/server/0431-Implement-Mob-Goal-API.patch (hunk 2 above)
Comment (tahmid-23) [reply patches/server/0431-Implement-Mob-Goal-API.patch]: sorry I need to figure out these settings
anyway the goal key should not be for skeletons it should be for abstractskeletons

Comment (Proximyst) [reply patches/server/0431-Implement-Mob-Goal-API.patch]: Go to your IntelliJ settings, search for "Class count to" and find these two options under `Editor -> Code Style -> Java -> Imports`:
image
Set the value to 999 or something stupidly high.

What does it actually fix, though? Can e.g. strays do melee attacks?

Comment (tahmid-23) [reply patches/server/0431-Implement-Mob-Goal-API.patch]: sorry about the intelliJ stuff thank you
anyway, wither skeletons use the same melee goal as in abstractskeleton, but if it's considered to be a skeleton goal then we can't apply it to wither skeletons.

**Review by Machine-Maker — APPROVED** [review]

**Review by Proximyst — APPROVED** [review]

File: patches/api/0203-Add-Mob-Goal-API.patch [hunk 3]
```diff
@@ -375,7 +375,7 @@ index 0000000000000000000000000000000000000000..659193fc0596084031c09aa47fbb428a
 +    GoalKey<Llama> TRADER_LLAMA_DEFEND_WANDERING_TRADER = GoalKey.of(Llama.class, NamespacedKey.minecraft("trader_llama_defend_wandering_trader"));
 +    GoalKey<Wither> WITHER_DO_NOTHING = GoalKey.of(Wither.class, NamespacedKey.minecraft("wither_do_nothing"));
 +    GoalKey<Illager> RAIDER_OPEN_DOOR = GoalKey.of(Illager.class, NamespacedKey.minecraft("raider_open_door"));
-+    GoalKey<Skeleton> SKELETON_MELEE = GoalKey.of(Skeleton.class, NamespacedKey.minecraft("skeleton_melee"));
++    GoalKey<AbstractSkeleton> SKELETON_MELEE = GoalKey.of(AbstractSkeleton.class, NamespacedKey.minecraft("abstract_skeleton_melee"));
```
Comment (qixils) [inline patches/api/0203-Add-Mob-Goal-API.patch:378]: why rename this? also, shouldn't MobGoalHelper L217 be updated as well?
```java
deobfuscationMap.put("abstract_skeleton_1", "skeleton_melee");
```

Comment (qixils) [reply patches/api/0203-Add-Mob-Goal-API.patch:378]: nvm, you did rename it, i swear i double checked. still curious as to why the rename is necessary

**Pull request merged**